client/engine/store/mockstore_test.go

```diff
@@ -1,91 +1,69 @@
-package store
+package store_test
 
 import (
 	"testing"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/google/go-cmp/cmp"
-	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/client/engine/store"
 	nc "github.com/statechannels/go-nitro/crypto"
+	td "github.com/statechannels/go-nitro/internal/testdata"
 	"github.com/statechannels/go-nitro/protocols"
-	"github.com/statechannels/go-nitro/protocols/directfund"
 )
 
 func TestNewMockStore(t *testing.T) {
 	sk := common.Hex2Bytes(`2af069c584758f9ec47c4224a8becc1983f28acfbe837bd7710b70f9fc6d5e44`)
-	NewMockStore(sk)
+	store.NewMockStore(sk)
 }
 
 func TestSetGetObjective(t *testing.T) {
 	sk := common.Hex2Bytes(`2af069c584758f9ec47c4224a8becc1983f28acfbe837bd7710b70f9fc6d5e44`)
 
-	ms := NewMockStore(sk)
+	ms := store.NewMockStore(sk)
 
 	id := protocols.ObjectiveId("404")
 	got, err := ms.GetObjectiveById(id)
 	if err == nil {
 		t.Errorf("expected not to find the %s objective, but found %v", id, got)
 	}
 
-	ts := state.TestState
-	ts.TurnNum = 0
-	request := directfund.ObjectiveRequest{
-		MyAddress:         ts.Participants[0],
-		CounterParty:      ts.Participants[1],
-		AppData:           ts.AppData,
-		AppDefinition:     ts.AppDefinition,
-		ChallengeDuration: ts.ChallengeDuration,
-		Nonce:             ts.ChannelNonce.Int64(),
-		Outcome:           ts.Outcome,
-	}
-	testObj, _ := directfund.NewObjective(request, false)
+	want := td.Objectives.Directfund.GenericDFO
 
-	if err := ms.SetObjective(&testObj); err != nil {
-		t.Errorf("error setting objective %v: %s", testObj, err.Error())
+	if err := ms.SetObjective(&want); err != nil {
+		t.Errorf("error setting objective %v: %s", want, err.Error())
 	}
 
-	got, err = ms.GetObjectiveById(testObj.Id())
+	got, err = ms.GetObjectiveById(want.Id())
 
 	if err != nil {
 		t.Errorf("expected to find the inserted objective, but didn't: %s", err)
 	}
 
-	if got.Id() != testObj.Id() {
+	if got.Id() != want.Id() {
 		t.Errorf("expected to retrieve same objective Id as was passed in, but didn't")
 	}
 }
 
 func TestGetObjectiveByChannelId(t *testing.T) {
 	sk := common.Hex2Bytes(`2af069c584758f9ec47c4224a8becc1983f28acfbe837bd7710b70f9fc6d5e44`)
 
-	ms := NewMockStore(sk)
-
-	ts := state.TestState
-	ts.TurnNum = 0
-	request := directfund.ObjectiveRequest{
-		MyAddress:         ts.Participants[0],
-		CounterParty:      ts.Participants[1],
-		AppData:           ts.AppData,
-		AppDefinition:     ts.AppDefinition,
-		ChallengeDuration: ts.ChallengeDuration,
-		Nonce:             ts.ChannelNonce.Int64(),
-		Outcome:           ts.Outcome,
-	}
-	testObj, _ := directfund.NewObjective(request, false)
+	ms := store.NewMockStore(sk)
+
+	want := td.Objectives.Directfund.GenericDFO
```

Comment:
As I understand it, this is the result of a constructor call which is only called once (not entirely sure when, perhaps at compile time?). If I write another test and reuse this generic direct funding objective, is there a risk that the two instances have pointers to the same values, which could cause problems if we go about mutating those values during the tests?

Might it be better to expose the constructor function `genericDFO()` itself, and call that function from the test to ensure we get a fresh objective?

Reply:
This is in fact what I had originally written, but refactored to replace the "complicated" function type with a concrete object. The generator is strictly better for the reason you've described.

fixed in 4033a4f